src/amo/components/UserProfile/index.js

```diff
@@ -0,0 +1,170 @@
+import Card from 'ui/components/Card';
+import DefinitionList, { Definition } from 'ui/components/DefinitionList';
+import LoadingText from 'ui/components/LoadingText';
+import Rating from 'ui/components/Rating';
+import UserAvatar from 'ui/components/UserAvatar';
+import type { UsersStateType, UserType } from 'amo/reducers/users';
+import type { DispatchFunc } from 'core/types/redux';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { I18nType } from 'core/types/i18n';
+
+import './styles.scss';
+
+
+type UserProfileParams = {| username: string |};
+
+type Props = {|
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+  params: UserProfileParams,
+  user: UserType,
+|};
+
+export class UserProfileBase extends React.Component<Props> {
+  componentWillMount() {
+    const { dispatch, errorHandler, params, user } = this.props;
+
+    if (!user) {
+      dispatch(fetchUserAccount({
+        errorHandlerId: errorHandler.id,
+        username: params.username,
+      }));
+    }
+  }
+
+  componentWillReceiveProps({ params: newParams }: { params: UserProfileParams }) {
+    const { dispatch, errorHandler, params: oldParams } = this.props;
+
+    if (oldParams.username !== newParams.username) {
+      dispatch(fetchUserAccount({
+        errorHandlerId: errorHandler.id,
+        username: newParams.username,
+      }));
+    }
+  }
+
+  render() {
+    const {
+      errorHandler,
+      i18n,
+      params,
+      user,
+    } = this.props;
+
+    if (errorHandler.hasError()) {
+      log.warn('Captured API Error:', errorHandler.capturedError);
+
+      if (errorHandler.capturedError.responseStatusCode === 404) {
+        return <NotFound errorCode={errorHandler.capturedError.code} />;
+      }
+    }
+
+    const userProfileHeader = (
+      <React.Fragment>
+        <UserAvatar className="UserProfile-avatar" user={user} />
+
+        <h1 className="UserProfile-name">
+          {user ? user.displayName : <LoadingText />}
+        </h1>
+      </React.Fragment>
+    );
+    const userProfileTitle = i18n.sprintf(
+      i18n.gettext('User Profile for %(user)s'), {
+        user: user ? user.displayName : params.username,
+      }
+    );
+
+    return (
+      <div className="UserProfile">
+        <Helmet>
+          <title>{userProfileTitle}</title>
+        </Helmet>
+
+        {errorHandler.renderErrorIfPresent()}
+
+        <div className="UserProfile-wrapper">
+          <Card
+            className="UserProfile-user-info"
+            header={userProfileHeader}
+          >
+            <DefinitionList className="UserProfile-dl">
+              {user && user.homepage ? (
+                <Definition
+                  className="UserProfile-homepage"
+                  term={i18n.gettext('Homepage')}
+                >
+                  {/* TODO: Make a helper that strips https?:\/\/ from text */}
+                  <a href={user.homepage}>{user.homepage.replace(/^https?:\/\//, '')}</a>
```

Comment:
Well, huh, I'm not sure how Flow lets you call `.replace()`. `user.homepage` can be `null` so I think you need protect against that.

Reply:
This code is only executed if `user && user.homepage` is truthy, so that's unlikely, but I will tidy this up regardless.